core/commands/storage.go

```diff
@@ -995,7 +998,7 @@ var storageUploadRecvContractCmd = &cmds.Command{
 func payWithSigning(ctx context.Context, req *cmds.Request, cfg *config.Config, n *core.IpfsNode, env cmds.Environment,
 	ss *storage.FileContracts, contractRequest *escrowpb.EscrowContractRequest) error {
 	// collecting all signed contracts means init status finished
-	ss.SendSessionStatusChan(storage.InitStatus, true, nil)
+	ss.SendSessionStatusChanPerMode(storage.InitStatus, true, nil, ctx)
```

Comment:
I don't feel comfortable passing `ctx` around. Couldn't you try to set it on a `ss` field for this particular state?

Reply:
fixed